sql/src/main/java/org/apache/druid/sql/calcite/schema/MetadataSegmentView.java

```diff
@@ -97,6 +101,9 @@ public MetadataSegmentView(
     this.isCacheEnabled = config.isMetadataSegmentCacheEnable();
     this.pollPeriodInMS = config.getMetadataSegmentPollPeriod();
     this.scheduledExec = Execs.scheduledSingleThreaded("MetadataSegmentView-Cache--%d");
+    this.segmentIdToReplicationFactor = CacheBuilder.newBuilder()
+                                                    .expireAfterWrite(3, TimeUnit.MINUTES)
```

Comment:
This can a more actually. Maybe 10 minutes?  And should it be expireAfterRead()?

Reply:
Is there a reason to prefer expireAfterRead? In the case that the coordinator for some reason sends only -1 for 30 minutes, in the case of `expireAfterRead` we would never start returning -1. In case of expireAfterWrite, after 10 minutes, we would return it. I think the second behaviour is better, wdyt?

Reply:
If the coordinator is not setting the RF factor to null for whatever reason, we shouldn't remove the entry from the cache since coordinator is yet to compute the RF factor. Hence is should expireAfterRead(10M).

Reply:
Changed